cmd/docgen/main.go

```diff
@@ -12,18 +13,61 @@ import (
 )
 
 func main() {
-	if len(os.Args) != 2 {
-		log.Fatal("Usage: cmd/docgen /path/to/outdir")
-	}
+	cmd := &cobra.Command{Use: "docgen"}
 
 	pluginSystem := system.NewPluginSystem()
 	if err := pluginSystem.Configure(ioutils.DefaultStreams); err != nil {
 		log.Fatal(err)
 	}
 	defer pluginSystem.TearDown()
 
-	err := doc.GenMarkdownTree(root.NewDefaultRootCmd(pluginSystem), os.Args[1])
-	if err != nil {
+	aspectRootCmd := root.NewDefaultRootCmd(pluginSystem)
+
+	cmd.AddCommand(NewCommandListCmd(aspectRootCmd))
+	cmd.AddCommand(NewGenMarkdownCmd(aspectRootCmd))
+
+	if err := cmd.Execute(); err != nil {
 		log.Fatal(err)
 	}
 }
+
+func NewCommandListCmd(aspectRootCmd *cobra.Command) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "command-list",
+		Short: "Creates a .bzl file with the top-level list of commands of the aspect CLI.",
+		RunE: func(_ *cobra.Command, _ []string) (exitErr error) {
+			fmt.Println(`"""Generated file - do NOT edit!`)
+			fmt.Println("This module contains the list of top-level commands from the aspect CLI.")
+			fmt.Println(`"""`)
+			fmt.Println("COMMAND_LIST = [")
+			cmds := aspectRootCmd.Commands()
+			for _, cmd := range cmds {
+				if cmd.IsAvailableCommand() {
+					fmt.Printf("    %q,\n", cmd.Use)
+				}
+			}
+			fmt.Println("]")
+			return nil
+		},
+	}
+	return cmd
+}
+
+func NewGenMarkdownCmd(aspectRootCmd *cobra.Command) *cobra.Command {
+	var outputDir string
+
+	cmd := &cobra.Command{
+		Use:   "gen-markdown",
+		Short: "Generates the markdown documentation.",
+		RunE: func(_ *cobra.Command, _ []string) error {
+			if outputDir == "" {
+				return fmt.Errorf("missing --output-dir")
```

Comment:
Doesn't cobra do this for you? As in, marking a flag as required?

Reply:
Right... I tried `MarkFlagRequired` and since it didn't work I thought 3 lines of code was just easier in this case. I took a look again and it wants `MarkPersistentFlagRequired`.

Fixed in 24ef379.